### PR TITLE
Authorized keys custom path

### PR DESCRIPTION
**Pull request by lubomir-kacalek**
Hi,
this propposed change allow to have configured a custom authorized key file path for an ssh server.
Thanks.
Lubomir Kacalek

### CODE REVIEW

File: templates/default/opensshd.conf.erb [hunk 1]
```diff
@@ -96,6 +96,11 @@ MaxStartups 10:30:100
 # Enable public key authentication
 PubkeyAuthentication yes
 
+<% if @node['ssh-hardening']['ssh']['server']['authorized_keys_path'] != nil %>
```
Comment (artem-sidorenko) [inline templates/default/opensshd.conf.erb]: you do not need `!= nil` here

**Comment by artem-sidorenko** [comment]
@lubomir-kacalek I would like to release the next minor version after #173 is merged, do you think you will be able to update this PR in the next days, so we can include it in the release?

**Comment by artem-sidorenko** [comment]
@lubomir-kacalek as discussed, the tests might look like this (in the `spec/recipes/server_spec.rb`)

```ruby
describe 'customized AuthorizedKeysFile option' do
  context 'without customized AuthorizedKeysFile' do
      cached(:chef_run) do
        ChefSpec::ServerRunner.new.converge(described_recipe)
      end

      it 'does not have AuthorizedKeysFile configured' do
        expect(chef_run).not_to render_file('/etc/ssh/sshd_config').
          with_content(/^AuthorizedKeysFile/)
      end
  end

  context 'with customized AuthorizedKeysFile' do
      cached(:chef_run) do
        ChefSpec::ServerRunner.new do |node|
          node.normal['ssh-hardening']['ssh']['server']['authorized_keys_path'] = '/some/authorizedkeysfile'
        end.converge(described_recipe)
      end

      it 'has AuthorizedKeysFile configured' do
        expect(chef_run).to render_file('/etc/ssh/sshd_config').
          with_content(/^AuthorizedKeysFile /some/authorizedkeysfile/)
      end
  end
end
```

**Comment by artem-sidorenko** [comment]
@lubomir-kacalek its also possible to have it a bit simplier:

```ruby
with_content('AuthorizedKeysFile /some/authorizedkeysfile')
```

instead of 
```ruby
with_content(/^AuthorizedKeysFile /some/authorizedkeysfile/)
```

and 

```ruby
with_content('AuthorizedKeysFile')
```

instead of
```ruby
with_content(/^AuthorizedKeysFile/)
```

**Comment by atomic111** [comment]
@lubomir-kacalek can you please sign your pr?

**Comment by artem-sidorenko** [comment]
@lubomir-kacalek @atomic111 means to use the sign-off, e.g. `git commit -s` (see the man page for details)

**Comment by lubomir-kacalek** [comment]
Hi @artem-sidorenko,
required changes has been pushed to my fork. Thank you in advance for support with testing part :-)

@atomic111 : New commit has been signed off as well.

Best regards,
   
       Lubos

**Comment by artem-sidorenko** [comment]
@lubomir-kacalek I looks better, I resolved the conflicts to master

Can you please make the rubocop happy? [There are some](https://travis-ci.org/dev-sec/chef-ssh-hardening/jobs/221815768) offenses

**Comment by artem-sidorenko** [comment]
@atomic111 any remarks to this PR?

I would like to merge it when rubocop is made happy and release the 2.1 of ssh-hardening

**Comment by artem-sidorenko** [comment]
@lubomir-kacalek thank you!

2.1 will be released today/tomorrow

**Pull request merged**